File: src/types/listing.ts
import { Timestamp } from '@react-native-firebase/firestore';

export type Category =
  | 'electronics'
  | 'books'
  | 'clothing'
  | 'dorm_stuff'
  | 'utilities'
  | 'sports'
  | 'tickets'
  | 'other';

export type Condition = 'new' | 'like_new' | 'good' | 'fair';

export type ListingStatus = 'active' | 'sold' | 'reserved' | 'removed';

export interface Listing {
  id: string;
  sellerId: string;
  sellerName: string;
  sellerAvatarUrl: string | null;
  sellerRating: number;
  campusId: string;
  title: string;
  description: string;
  price: number;
  isDonate: boolean;
  category: Category;
  condition: Condition;
  images: string[];
  aiPriceSuggestion: number | null;
  status: ListingStatus;
  viewCount: number;
  createdAt: Timestamp;
  updatedAt: Timestamp;
}

export interface WishlistItem {
  id: string;
  query: string;
  category: Category | null;
  maxPrice: number | null;
  createdAt: Timestamp;
  notifiedListings: string[];
}

export interface WishlistIndex {
  id: string;
  userId: string;
  campusId: string;
  category: Category | null;
  maxPrice: number | null;
  query: string;
  fcmToken: string | null;
}

File: lib/models/listing.dart
class Listing {
  final String id;
  final String sellerId;
  final String sellerName;
  final String? sellerAvatarUrl;
  final double sellerRating;
  final String title;
  final String description;
  final double price;
  final bool isDonate;
  final String category;
  final String condition;
  final List<String> images;
  final DateTime createdAt;

  const Listing({
    required this.id,
    required this.sellerId,
    required this.sellerName,
    this.sellerAvatarUrl,
    this.sellerRating = 0,
    required this.title,
    required this.description,
    required this.price,
    this.isDonate = false,
    required this.category,
    required this.condition,
    required this.images,
    required this.createdAt,
  });

  String get formattedPrice => isDonate ? 'FREE' : '\$${price.toStringAsFixed(price % 1 == 0 ? 0 : 2)}';
}

class WishlistItem {
  final String id;
  final String query;
  final String? category;
  final double? maxPrice;

  const WishlistItem({
    required this.id,
    required this.query,
    this.category,
    this.maxPrice,
  });
}
